### DISCUSSION TITLE
Resources on multiple ports

### DISCUSSION THREAD

**Discussion by phani50** [dropwizard/dropwizard#6983] [Q&A]
we are using dropwizard for the microservices. we have a requirement where we have to expose two http ports.
lets say few resources (rest apis) should exposed/served by one http port and few resouses by other port.
Currently i don't see a way how to distribute resources between multiple http ports/endpoins

**Comment by zUniQueX** [comment]
Hi @phani50. The Jetty server allows using multiple ports and this can already be configured with the `DefaultServerFactory`. But we register only one Jersey servlet for the REST modelling.

If you want to distribute resources between ports, you would either need to use a second Jersey servlet where you register the resources on either one or you could achieve this with some kind of authentication. There you would abort execution of specific endpoints based on the given port.

**Reply by phani50** [reply]
Hi @zUniQueX ,
Thanks for the reply, yes i have already tried **DefaultServerFactory**, as you mentioned all resources are available for both ports here.

How to use (or) get second Jersey servlet ? Is there any example for this ?

**Reply by zUniQueX** [reply]
I'm not aware of example code for this as I assume it's a very rare use case.

You'd have to override the `DefaultServerFactory` to provide a second servlet like it's done in the `createAppServlet` method. Access to this servlet for registering resources has to be provided through the environment. So you'd also have to override the `EnvironmentCommand#run(Bootstrap, Namespace, T)` method to provide a custom `Environment`. In the overridden server factory, you would register the second `JerseyEnvironment` there.
